sdk/eventhub/azure-eventhubs/azure/eventhub/common.py

```diff
@@ -258,6 +258,9 @@ def body_as_json(self, encoding='UTF-8'):
         except Exception as e:
             raise TypeError("Event data is not compatible with JSON type: {}".format(e))
 
+    def encode_message(self):
+        return self.message.encode_message()
+
```

Comment:
Do we want to make this public? What's the scenario requiring this?

Reply:
This is apparently the "plugin" required by the underlying UAMQP bits that we brought forward to support this scenario. (was stolen wholesale from Adam's track2 PR that fixes this bug)